File: SuperiorPackGroup/BLL/ReportsBLL.vb
Imports SuperiorPackGroup.SPGTableAdapters

Public Class ReportsBLL

    Private m_ReportsTableAdapter As ReportsTableAdapter = Nothing

    Public ReadOnly Property Adapter() As ReportsTableAdapter
        Get
            If (Me.m_ReportsTableAdapter Is Nothing) Then
                Me.m_ReportsTableAdapter = New ReportsTableAdapter
                Me.m_ReportsTableAdapter.Connection.ConnectionString = My.Settings.UserConnectionString
            End If
            Return Me.m_ReportsTableAdapter
        End Get
    End Property

    Public Function GetReports() As SPG.ReportsDataTable

        Return Adapter.GetReports()

    End Function

    Public Function GetReportsByModule(ByVal ModuleName As String) As SPG.ReportsDataTable

        Return Adapter.GetReportsByModule(ModuleName)

    End Function

End Class

File: SuperiorPackGroup/BLL/ReportsBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using SuperiorPackGroup.SPGTableAdapters;

namespace SuperiorPackGroup
{
	public class ReportsBLL
	{

		private ReportsTableAdapter m_ReportsTableAdapter = null;

		public ReportsTableAdapter Adapter
		{
			get
			{
				if (m_ReportsTableAdapter == null)
				{
					m_ReportsTableAdapter = new ReportsTableAdapter();
					m_ReportsTableAdapter.Connection.ConnectionString = Properties.Settings.Default.UserConnectionString;
				}
				return m_ReportsTableAdapter;
			}
		}

		public SPG.ReportsDataTable GetReports()
		{

			return Adapter.GetReports();

		}

		public SPG.ReportsDataTable GetReportsByModule(string ModuleName)
		{

			return Adapter.GetReportsByModule(ModuleName);

		}

	}

}